packages/styled-components/src/utils/hyphenateStyleName.js

```diff
@@ -5,8 +5,10 @@
  * https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/core/hyphenateStyleName.js
  */
 
-const uppercasePattern = /([A-Z])/g;
+const uppercaseCheck = /([A-Z])/;
+const uppercasePattern = new RegExp(uppercaseCheck, 'g');
```

Comment:
Why this instead of just adding `/g` to `uppercaseCheck`?

Reply:
For performance reasons it's using `Regexp.prototype.test` (as opposed to String.prototype.match) and quoting [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test):
> As with exec() (or in combination with it), test() called multiple times on the same global regular expression instance will advance past the previous match.

It's a little unclear how it works so I've always avoided it - is each regex storing which strings it's been matched against?
Or is it a per-regex counter for all strings it's matched against?
```
let str1 = 'fooBarBaz'
let str2 = 'howAboutNow'
let regex = /([A-Z])/g
> regex.test(str1)
true
> regex.test(str1)
true
> regex.test(str1)
false
> regex.test(str1)
true
> regex.test(str2)
true
> regex.test(str1)
false
```
This seems to imply it's a per-regex counter, so by using a non-global version, the test check isn't internally changing its state.

As further proof of this, changing `uppercaseCheck` to `/([A-Z])/g` causes a load of tests to start failing as some names stop becoming hyphenated!

Reply:
Hmm, fair enough.